src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt

```diff
@@ -41,88 +41,123 @@ class ApptentiveKit : KitIntegration(), KitIntegration.EventListener, CommerceLi
 
         if (apptentiveAppKey != null && apptentiveAppSignature != null) {
             val configuration = ApptentiveConfiguration(apptentiveAppKey, apptentiveAppSignature)
-            Apptentive.register(context.applicationContext as Application, configuration)
-            ApptentiveNotificationCenter.defaultCenter()
-                .addObserver(
-                    ApptentiveNotifications.NOTIFICATION_CONVERSATION_STATE_DID_CHANGE,
-                    this
-                )
+            configuration.logLevel = getApptentiveLogLevel()
+            configuration.distributionVersion = com.mparticle.BuildConfig.VERSION_NAME
+            configuration.distributionName = "mParticle"
+            configuration.shouldSanitizeLogMessages =
+                StringUtils.tryParseSettingFlag(settings, SHOULD_SANITIZE_LOG_MESSAGES, true)
+            configuration.shouldEncryptStorage =
+                StringUtils.tryParseSettingFlag(settings, SHOULD_ENCRYPT_STORAGE, false)
+            configuration.shouldInheritAppTheme =
+                StringUtils.tryParseSettingFlag(settings, SHOULD_INHERIT_APP_THEME, true)
+            configuration.customAppStoreURL = settings[CUSTOM_APP_STORE_URL]
+            configuration.ratingInteractionThrottleLength = StringUtils.tryParseLongSettingFlag(
+                settings,
+                RATING_INTERACTION_THROTTLE_LENGTH,
+                TimeUnit.DAYS.toMillis(7)
+            )
+            Apptentive.register(
+                context.applicationContext as Application,
+                configuration
+            ) { registerResult ->
+                if (registerResult is RegisterResult.Success) {
+                    Apptentive.setMParticleId(currentUser?.id.toString())
+                }
+            }
         }
         return emptyList()
     }
 
-    override fun onKitDestroy() {
```

Comment:
Why should we remove the onKitDestroy block which dettacches this observer from Apptentive?

Reply:
The observers were no longer used/available in the new SDK. This wasn't needed.